src/engine/ExecuteUpdate.cpp

```diff
@@ -176,6 +176,18 @@ ExecuteUpdate::computeGraphUpdateQuads(
       cancellationHandle->throwIfCancelled();
     }
   }
+  auto sortAndRemoveDuplicates = [](auto& vec) {
+    ql::ranges::sort(vec);
+    vec.erase(std::unique(vec.begin(), vec.end()), vec.end());
+  };
+  sortAndRemoveDuplicates(toInsert);
+  sortAndRemoveDuplicates(toDelete);
+  metadata.inUpdate = DeltaTriplesCount{static_cast<int64_t>(toInsert.size()),
+                                        static_cast<int64_t>(toDelete.size())};
+  std::vector<IdTriple<>> reducedToDelete;
+  ql::ranges::set_difference(std::move(toDelete), toInsert,
+                             std::back_inserter(reducedToDelete));
+  toDelete.swap(reducedToDelete);
```

Comment:
This can be factored out into a small and static function , that can then be documented and tested.